File: 프로그래머스/2/42587. 프로세스/프로세스.py
def solution(priorities, location):
    priorities = [[x, False] for x in priorities]
    priorities[location][1] = True
    count = 1
    
    while True:
        target = priorities[0]
        check_items = [item for item in priorities[1:] if target[0] < item[0]]
        if check_items:
            item = priorities.pop(0)
            priorities.append(item)
        else:
            if target[1]:
                return count
            else:
                item = priorities.pop(0)
                count += 1

File: 프로그래머스/2/42587. 프로세스/프로세스.js
function solution(priorities, location) {
    let answer = 1;
    
    priorities = priorities.map((item, index) => [item, index]);
    
    while (true) {
        const [priority, position] = priorities.shift();
        
        // 우선순위가 더 높은 프로세스 여부 확인
        if (priorities.filter((item) => item[0] > priority).length) {
            priorities.push([priority, position]);
        } else {
            // 목표 location인지 확인
            if (position === location) {
                break;
            } else {
                answer++;
            }
        }
    }
    
    return answer;
}